In [1]:
import numpy as np

#### Question 1
Given a space separated list of numbers.
We have to print a reversed NumPy array with the element type float.

In [9]:
lst = [1,2,3,4,5]
arr = np.array(lst,float)
arr = np.flipud(arr)
arr = arr.astype(int)
arr

array([5, 4, 3, 2, 1])

#### Question 2
Given a space separated list of nine integers. We have to convert this list into a 3X3 NumPy array.

In [13]:
#output of the shape method of theype (m,n) = (rows, colomns)
lst1 = [ 1,2,3,4,5,6,7,8,9]
arr1 = np.array(lst1)
print(np.reshape(arr1,(3,3)))

[[1 2 3]
 [4 5 6]
 [7 8 9]]


#### Question 3
Remove rows in Numpy array that contains non-numeric values.

In [18]:
arr = np.array([[10.5, 22.5, 3.8, 5],
                  [23.45, 50, 78.7, 3.5],
                  [41, np.nan, np.nan, 0],
                  [20, 50.20, np.nan, 2.5],
                  [18.8, 50.60, 8.8, 58.6]])
print(arr.shape)
print(arr[~np.isnan(arr).any(axis=1)])

(5, 4)
[[10.5  22.5   3.8   5.  ]
 [23.45 50.   78.7   3.5 ]
 [18.8  50.6   8.8  58.6 ]]


#### Question 4
Find the most frequent value in a NumPy array.

In [27]:
lst4 = [1,2,324,24,4,2,1,2,23,43413,12313,23,12323,34]
dc = {}
for i in lst4:
    if i in dc:
        dc[i] +=1
    else:
        dc[i] = 1

print(max(dc, key = dc.get))

2


In [28]:
print(np.bincount(np.array(lst4)).argmax())

2


#### Question 5
Find the eigen values of a matrix.

A =[[10.5  22.5   3.8]
    [23.45 50.   78.7]
    [18.8  50.6   8.8]]
det(A -lambdaxI) = 0

In [29]:
from numpy import linalg as la

In [30]:
eg = la.eigvals([[1,2,3,4],[5,6,7,8],[9,10,11,12],[13,14,15,16]])
print(eg)

[ 3.62093727e+01 -2.20937271e+00 -3.18863232e-15 -1.34840081e-16]
